src/rgw/rgw_user.cc

```diff
@@ -1208,7 +875,6 @@ int RGWAccessKeyPool::execute_remove(RGWUserAdminOpState& op_state, std::string
     return -ERR_INVALID_ACCESS_KEY;
   }
 
-  rgw_remove_key_index(store, kiter->second);
```

Comment:
these calls to rgw_remove_key_index() were all removed. it looks like the responsibility to delete them was moved into `store_user_info()`, but its PutOperation::remove_old_indexes() didn't handle the access key objects. i added a commit to do this; does that looks right @yehudasa?

Reply:
@cbodley yup!